// @adobe/coral-spectrum/coral-component-steplist/src/templates/step.js
const template = (function anonymous(data_0
) {
  var frag = document.createDocumentFragment();
  var data = data_0;
  var el0 = this["link"] = document.createElement("a");
  el0.className += " _coral-Steplist-link";
  el0.setAttribute("handle", "link");
  var el1 = document.createTextNode("\n  ");
  el0.appendChild(el1);
  var el2 = this["accessibilityLabel"] = document.createElement("span");
  el2.className += " u-coral-screenReaderOnly _coral-Steplist-accessibilityLabel";
  el2.setAttribute("handle", "accessibilityLabel");
  el0.appendChild(el2);
  var el3 = document.createTextNode("\n  ");
  el0.appendChild(el3);
  var el4 = this["stepMarkerContainer"] = document.createElement("span");
  el4.className += " _coral-Steplist-markerContainer";
  el4.setAttribute("handle", "stepMarkerContainer");
  el4.setAttribute("role", "img");
  el4.setAttribute("aria-hidden", "true");
  var el5 = document.createTextNode("\n    ");
  el4.appendChild(el5);
  var el6 = document.createElement("span");
  el6.className += " _coral-Steplist-marker";
  el4.appendChild(el6);
  var el7 = document.createTextNode("\n  ");
  el4.appendChild(el7);
  el0.appendChild(el4);
  var el8 = document.createTextNode("\n  ");
  el0.appendChild(el8);
  var el9 = this["overlay"] = document.createElement("coral-tooltip");
  el9.setAttribute("tracking", "off");
  el9.setAttribute("smart", "");
  el9.setAttribute("focusonshow", "off");
  el9.setAttribute("handle", "overlay");
  el9.setAttribute("placement", "top");
  el9.setAttribute("variant", "inspect");
  el9.setAttribute("interaction", "off");
  el9.setAttribute("breadthoffset", "1");
  el0.appendChild(el9);
  var el10 = document.createTextNode("\n");
  el0.appendChild(el10);
  frag.appendChild(el0);
  var el11 = document.createTextNode("\n");
  frag.appendChild(el11);
  var el12 = this["line"] = document.createElement("span");
  el12.className += " _coral-Steplist-segment";
  el12.setAttribute("handle", "line");
  frag.appendChild(el12);
  var el13 = document.createTextNode("\n");
  frag.appendChild(el13);
  return frag;
});
export default template;
